guides/v2.1/howdoi/patch_from_github.md

```diff
@@ -0,0 +1,62 @@
+---
+layout: default
+group: howdoi
+title: Create a GitHub patch for a Composer installation
+version: 2.2
+github_link: howdoi/patch_from_github.md
+functional_areas:
+  - Install
+  - System
+  - Setup
+---
+
+Sometimes it takes a while for us to include a bug fix made on GitHub in a Magento 2 Composer release. In the meantime, you can create a patch from GitHub and use the `composer-patches` plugin to apply it to your Composer-based Magento 2 installation.
+
+<div class="bs-callout bs-callout-warning" markdown="1">
+Be careful applying an unreleased patch. Do comprehensive testing before deploying it.
+</div>
+
+## Create a patch
+
+To create a patch file from a GitHub commit or pull request, append `.patch` to the url, [https://github.com/magento/magento2/commit/2d31571f1bacd11aa2ec795180abf682e0e9aede.patch](https://github.com/magento/magento2/commit/2d31571f1bacd11aa2ec795180abf682e0e9aede.patch).
+
+<div class="bs-callout bs-callout-info" markdown="1">
```

Comment:
Give an example of what needs to be changed. Would you change every instance of `app/code/Magento/` to `vendor/Magento`?

Reply:
@dmanners, deferring to you on this

Reply:
Indeed, example would be very useful here.

BTW: the `cweagans/composer-patches` package's original intent was to apply patches relative to the root of the package, not to the root of the application. That particular "feature" was added afterwards, but I'm considering it a hack and never used it. So we always change the paths to the relative path of the module itself.

So in this specific case, you would replace all instances (5 times) of `app/code/Magento/Payment/view/frontend/web/js/view/payment/iframe.js` with `view/frontend/web/js/view/payment/iframe.js`

(But if it would be considered easier to understand, then just replace `app/code/Magento/Payment/` with `vendor/magento/module-payment/` like was mentioned below on line 29)

Btw, maybe good to mention as a warning: you shouldn't make these changes with an actual code editor, since they sometimes strip out trailing whitespace or adding new lines depending on how they are configured, which can sometimes mess up the patch file. I prefer a very old school editor which doesn't do any fancy stuff like that. On macOS I'm using the built-in TextEdit application for this. For Windows, I'm assuming you can use the built-in Notepad? And for Linux, just vim or nano probably?